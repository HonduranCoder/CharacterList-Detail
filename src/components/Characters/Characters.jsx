import React from 'react';
import { Link } from 'react-router-dom';

export default function Characters({ characters }) {
  return (
    <ul>
      {characters.map((character) => (
        <li key={character._id}>
          <Link to={`/character/${character._id}`}>
            <div>
              <img
                src={character.photoUrl}
                alt={character.name}
                height="200"
                width="200"
              />
              <span>{character.name}</span>
            </div>
          </Link>
        </li>
      ))}
    </ul>
  );
}
