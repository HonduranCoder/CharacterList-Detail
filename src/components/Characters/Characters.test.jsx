import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Characters from './Characters.jsx';

it('should render Characters', () => {
  const characters = [
    {
      _id: '5cf5679a915ecad153ab68c8',
      position: 'Earth King',
      allies: 'Roayl Earthbender Guards',
    },
  ];
  const { container } = render(
    <MemoryRouter>
      <Characters characters={characters} />
    </MemoryRouter>
  );
  expect(container).toMatchSnapshot();
});
