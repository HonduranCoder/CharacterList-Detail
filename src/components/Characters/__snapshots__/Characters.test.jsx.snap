// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Characters 1`] = `
<div>
  <ul>
    <li>
      <a
        href="/character/5cf5679a915ecad153ab68c8"
      >
        <div>
          <img
            height="200"
            width="200"
          />
          <span />
        </div>
      </a>
    </li>
  </ul>
</div>
`;
